python_modules/dagster-graphql/dagster_graphql/schema/pipelines/pipeline.py

```diff
@@ -608,13 +613,23 @@ def resolve_repository(self, graphene_info):
         location = graphene_info.context.get_repository_location(handle.location_name)
         return GrapheneRepository(location.get_repository(handle.repository_name), location)
 
-    def resolve_assetNodes(self, graphene_info):
+    def resolve_assetNodes(self, graphene_info, **kwargs):
         from ..asset_graph import GrapheneAssetNode
 
         handle = self._external_pipeline.repository_handle
         location = graphene_info.context.get_repository_location(handle.location_name)
         repository = location.get_repository(handle.repository_name)
         asset_nodes = repository.get_external_asset_nodes(self._external_pipeline.name)
+
+        asset_keys = kwargs.get("assetKeys")
+        if asset_keys:
+            asset_keys = [AssetKey.from_graphql_input(asset_key) for asset_key in asset_keys]
+            filtered_asset_nodes = []
+            for asset_node in asset_nodes:
+                if asset_node.asset_key in asset_keys:
+                    filtered_asset_nodes.append(GrapheneAssetNode(repository, asset_node))
+            return filtered_asset_nodes
```

Comment:
nit: might be more "pythonic" to do the filtering in a list comprehension:

```
asset_keys = set(AssetKey.from_graphql_input(asset_key) for asset_key in kwargs.get("assetKeys", []))
filtered_asset_nodes = [asset_node for asset_node in asset_nodes if not asset_keys or asset_node.asset_key in asset_keys]
```

Reply:
👍  thanks!